### DISCUSSION TITLE
Using digest authentication for a camera

### DISCUSSION THREAD

**Discussion by emiltin** [bluenviron/mediamtx#2454] [Questions]
### Question

Hi,
We have a camera that only supports digest authentication, not basic.
Is it possible to configure mediamtx to use digest when connecting to the camera?

**Comment by aler9** [comment]
Hello, in RTSP, it's the camera that tells the client (in this case, MediaMTX) which authentication mechanisms are supported:

request from the client:

```
DESCRIBE rtsp://localhost:8554/test RTSP/1.0
Accept: application/sdp
CSeq: 3
```

reply from the camera:

```
RTSP/1.0 401 Unauthorized
CSeq: 3
WWW-Authenticate: Basic realm="IPCAM"
WWW-Authenticate: Digest realm="IPCAM", nonce="4a566b90d799e1bab1670d9d93f44183"
```

Thus if the camera supports digest only, the server already uses digest.

**Comment by emiltin** [comment]
Thank you for the clarification!

[a comment by an automated account is not shown]
